modules/hostap/src/supp_api.c

```diff
@@ -403,6 +404,19 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		}
 	}
 
+	memcpy((void *)&mac, params->bssid, WIFI_MAC_ADDR_LEN);
+	if (!net_eth_is_addr_broadcast(&mac) &&
```

Comment:
we should fail for invalid BSSID except for all zeroes (which can be treated as unset).

Reply:
Done